newdocs/typeclass-cheatsheet.md

```diff
@@ -0,0 +1,190 @@
+# Consensus Cheatsheet (Type Families, Classes, & Notes)
+
+See [Diagrammatic Conventions](#diagrammatic-conventions)
+    
+## Birds Eye Overview of Consensus Types
+### Type Families (standalone and associated types)
+    
+From inspecting the directions of the arrows, it should be clear that `b :: B`
+fully determines---directly or indirectly---all the other types.
+
+``` haskell
```

Comment:
suggestion: Use inline textual based diagram options provided by eg. mermaid or other tools out there.
Adding a `mermaid` tagged source code block allows GitHub to render it in the UI.
This will probably makes those nice diagrams more maintainable over time? Also, it's possible to include hyperlinks to relevant source code which would be even more helpful